purchase_order_line_schedule/models/purchase_order_line_schedule.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 ForgeFlow, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class PurchaseOrderLineSchedule(models.Model):
+
+    _name = "purchase.order.line.schedule"
+    _description = "Purchase Order Line Schedule"
+    _order = "order_line_id, date_planned, id"
+
+    order_id = fields.Many2one(
+        comodel_name="purchase.order", related="order_line_id.order_id", store=True
+    )
+    purchase_state = fields.Selection(related="order_line_id.order_id.state",)
+    qty_received_method = fields.Selection(related="order_line_id.qty_received_method",)
+    order_line_id = fields.Many2one(
+        comodel_name="purchase.order.line", required=True, ondelete="cascade"
+    )
+    date_planned = fields.Datetime(string="Scheduled Date", index=True, required=True)
```

Comment:
The operation cannot be completed:
- Create/update: a mandatory field is not set.
- Delete: another model requires the record being deleted. If possible, archive it instead.

Model: Purchase Order Line Schedule (purchase.order.line.schedule), Field: Scheduled Date (date_planned)

Should be required.